### PR TITLE
Add Replicas field to DeploymentParams struct

### PR DESCRIPTION
**Pull request by dharmit**
### What does this PR do?:

Adds `Replicas` field to `DeploymentParams` struct. This will help initialize a Deployment with Replica set to what the tooling would like to have. By default, library doesn't set it to anything at the moment.

### Which issue(s) this PR fixes:

https://github.com/redhat-developer/odo/issues/5347

### PR acceptance criteria:
Testing and documentation do not need to be complete in order for this PR to be approved. We just need to ensure tracking issues are opened.

> - Open new test/doc issues under the [devfile/api](https://github.com/devfile/api/issues) repo
> - Check each criteria if:
>  - There is a separate tracking issue. Add the issue link under the criteria
>  **or**
>  - test/doc updates are made as part of this PR
> -  If unchecked, explain why it's not needed

- [x] Unit/Functional tests

  

- [ ] [QE Integration test](https://github.com/devfile/integration-tests) 

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: pkg/devfile/generator/generators.go [hunk 1]
```diff
@@ -158,6 +159,7 @@ type DeploymentParams struct {
 	Containers        []corev1.Container
 	Volumes           []corev1.Volume
 	PodSelectorLabels map[string]string
+	Replicas          int32
```
Comment (feloy) [inline pkg/devfile/generator/generators.go:162]: I would set the Replicas as a pointer, to make it optional (the nil value indicating the user does not want to set it). The 0 value on which you test below is a valid value a user would want to set.

Comment (dharmit) [reply pkg/devfile/generator/generators.go]: Makes sense. Does https://github.com/devfile/library/pull/129/commits/145564efc02c37f573608a4875e81009f3da3427 look good?

File: pkg/devfile/generator/generators.go [hunk 2]
```diff
@@ -187,6 +187,10 @@ func GetDeployment(devfileObj parser.DevfileObj, deployParams DeploymentParams)
 		Spec:       *getDeploymentSpec(deploySpecParams),
 	}
 
+	if deployParams.Replicas != nil {
+		deployment.Spec.Replicas = deployParams.Replicas
```
Comment (feloy) [inline pkg/devfile/generator/generators.go:191]: I think that this affectation should be done in getDeploymentSpec which is the dedicated function for this.

Comment (dharmit) [reply pkg/devfile/generator/generators.go]: I agree. That's where I first looked. But that would mean changing the signature of `getDeploymentSpec`. I was not sure if that would be OK. But it's an internal function, so changing signature shouldn't be a big deal.

Pushed https://github.com/devfile/library/pull/129/commits/4a544b12a31ed89f58f1da0e08fa074ab203726a. PTAL.

File: pkg/devfile/generator/generators.go [hunk 3]
```diff
@@ -184,7 +184,7 @@ func GetDeployment(devfileObj parser.DevfileObj, deployParams DeploymentParams)
 	deployment := &appsv1.Deployment{
 		TypeMeta:   deployParams.TypeMeta,
 		ObjectMeta: deployParams.ObjectMeta,
-		Spec:       *getDeploymentSpec(deploySpecParams),
+		Spec:       *getDeploymentSpec(deploySpecParams, deployParams),
```
Comment (yangcao77) [inline pkg/devfile/generator/generators.go:187]: add it to be an additional field in `deploySpecParams` instead of changing the function input

**Review by yangcao77 — CHANGES_REQUESTED** [review]

File: pkg/devfile/generator/generators.go (hunk 3 above)
Comment (dharmit) [reply pkg/devfile/generator/generators.go]: @yangcao77 thanks, I have made the change.

@yangcao77 @feloy What surprised me was I no longer needed the `if` check. I was expecting a nil pointer error, but it seems that was my lack of understanding of Go.

**Comment by dharmit** [comment]
@yangcao77 do I need to make any changes to the PR? Would `go mod tidy` suffice?

**Comment by yangcao77** [comment]
@dharmit  yes, please run `go mod tidy` and `go fmt -x ./...`

**Review by yangcao77 — APPROVED** [review]

**Pull request merged**